### ISSUE TITLE
Thread watcher not working.

### ISSUE THREAD

**Issue by ccd0** [ccd0/4chan-x#1050]
Submitter: Aaron
Email: aaronstrauss1@gmail.com
Earlier I logged on and was browsing, it took a couple of minutes to notice that I had no threads in my thread watcher, which I found add as just a minute ago I had posted in a thread and I have "auto watch reply" enabled. So I went back to the thread and hit W to watch. I watched it appear in the thread watcher, then disappear, I went to other boards and it was relatively the same problem, though in some cases, I would hit W, and it would stay, then when I went back to the boards Catalog it would not be there. The steps to replicate said problem are very simple

1. Go to a thread
2. Hit W
3. Watch it pop up in your thread watcher and then either

A. Disappear immediately

or

B. Disappear when going to a new thread.

I have tested the problem by using 4chan's Native extension, and thread watching works fine on it, thus it seems to be a 4chanx problem only. Both Greasemonkey and 4chanx are up to date. Attached below is an imgur link to the picture of my Firefox Console. I notice "no element found" keeps appearing a lot, not sure if this is in relation to the error.

http://i.imgur.com/5hgnvto.png

I also have reset 4chanx's settings back to default, and the problem remains. 

As you can see from above, I have taken the necessary steps to prove that it is a 4chanx problem, and have given you a good amount of information on said problem. Hopefully you can fix it soon.

Thanks.

**Comment by ccd0** [comment]
I can't reproduce this.

Does it happen in a fresh Firefox profile with just Greasemonkey and 4chan X installed?

Are you running low on disk space?

Do you seen any errors in the console -- either the Javascript console (F12) or the browser console (Control+Shift+J) at the moment you watch the thread?

What specific version numbers (not just "latest") of Firefox / Greasemonkey / 4chan X do you have installed?

**Comment by Nusharik** [comment]
A fresh firefox profile as in disabling anything but Greasemonkey and 4chanx? If that is the case then yes, the error keeps occuring. As for disk space no, I have 526 gigabytes left. I noticed in my browser console there seems to be this error occuring in relation to me clicking W and choosing a new board (Which is when the watched thread is removed)

http://i.imgur.com/P7DA7Wp.png

In F12 I'm not sure how it works exactly but I did see these two errors.

http://i.imgur.com/np95vYS.png

My Firefox/Greasemonkey and 4chanX versions are as follows.

Firefox: 48.0.2 (Updated last night)

Greasemonkey: 3.9 (Last updated Sunday, August 21, 2016)

4chanx: 1.12.2.1 (I even installed it again earlier just in case)

**Comment by Nusharik** [comment]
I'm sorry, I just now realized what you meant by Firefox profile. Yes, I made a new Firefox profile just now and the error seems to be gone. Based on my images I sent you, do you know what seems to be causing the problem on my normal profile on firefox?

**Comment by ccd0** [comment]
>Based on my images I sent you, do you know what seems to be causing the problem on my normal profile on firefox?

I don't know. It could be some setting that you've set in your default profile, or it could just be a corrupt profile (happens sometimes).

**Comment by Nusharik** [comment]
Well the power did go out yesterday, that might have caused it to be corrupt I guess. I made a new profile, sync'd everything up and spent an hour opening all of my bookmarks so they could have the little thumbnail/icon picture next to them that tells you the site they are from, but everything is all good now and my thread watcher works. I guess the lesson here is for me to buy a battery backup. Thank you for your help ccd0, I've used your 4chanx for years and it's such a good extension. Thank you for your help, and I hope you have a good day.
